src/main/java/org/spongepowered/common/util/FireworkUtil.java

```diff
@@ -103,23 +116,54 @@ public static boolean removeFireworkEffects(final Object object) {
         return false;
     }
 
-    public static ItemStack getItem(final FireworkRocketEntity firework) {
-        ItemStack item = firework.getEntityData().get(FireworkRocketEntityAccessor.accessor$DATA_ID_FIREWORKS_ITEM());
+    public static boolean setFlightModifier(final FireworkRocketEntity firework, final int modifier) {
```

Comment:
I think it would be better to leave the item alone here and just change the fireworks lifetime unconditionally, this way you can change it for any item.